src/haproxy/conn.go

```diff
@@ -0,0 +1,126 @@
+package haproxy
+
+import (
+	"bufio"
+	"net"
+	"time"
+
+	proxyproto "github.com/pires/go-proxyproto"
+)
+
+// Conn is a `net.Conn` compatible struct that handles PROXY header checking.
+type Conn struct {
+	rd  *bufio.Reader
+	c   net.Conn
+	r   net.Addr
+	l   net.Addr
+	hdr *proxyproto.Header
+}
+
+// NewConn wraps a connection with a `bufio.Reader` for checking if the
+// PROXY protocol headers are present
+func NewConn(nc net.Conn) (net.Conn, error) {
+	c := &Conn{
+		c:  nc,
+		rd: bufio.NewReader(nc),
+	}
+	if err := c.ProxyInit(); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+// ProxyInit checks for the PROXY protocol headers, and if not returns
+// the connection unmolested.
+func (c *Conn) ProxyInit() error {
```

Comment:
Docs state that the function returns a connection, but it doesn't return anything.  I think this logic should be in an unexported method, or otherwise be part of `NewConn()`.

Reply:
Unexported is probably better, fixed.